src/lib/es2023.array.d.ts

```diff
@@ -20,7 +26,51 @@ interface Array<T> {
      * @param thisArg If provided, it will be used as the this value for each invocation of
      * predicate. If it is not provided, undefined is used instead.
      */
-    findLastIndex(predicate: (value: T, index: number, array: T[]) => unknown, thisArg?: any): number;
+    findLastIndex(
+        predicate: (value: T, index: number, array: T[]) => unknown,
+        thisArg?: any
+    ): number;
+
+    /**
+     * Copies the array and returns the copied array with all of its elements reversed.
+     */
+    toReversed(): T[];
+
+    /**
+     * Copies and sorts the array.
+     * @param compareFn Function used to determine the order of the elements. It is expected to return
+     * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
+     * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+     * ```ts
+     * [11, 2, 22, 1].toSorted((a, b) => a - b) // [1, 2, 11, 22]
+     * ```
+     */
+    toSorted(compareFn?: (a: T, b: T) => number): T[];
+
+    /**
+     * Copies an array and removes elements while, if necessary, inserting new elements in their place, returning the remaining elements.
+     * @param start The zero-based location in the array from which to start removing elements.
+     * @param deleteCount The number of elements to remove.
+     * @param items Elements to insert into the array in place of the deleted elements.
+     * @returns A copy of the original array with the remaining elements.
+     */
+    toSpliced<F>(start: number, deleteCount: number, ...items: F[]): (T | F)[];

```

Comment:
I am not sure about this. I believe that this is similar to the syntax of extending an array like with the spreading syntax. Also, I think that this might be useful when passing into libraries such as configs in which you want to introduce a new config option which wouldn't conform to the new type. Finally, I feel like there's really no need to restrict it to the old items' type when this is a whole new array.

Reply:
I still disagree, for a few reasons:
1. The default expectation for array methods is not to add types, even other pure ones like `concat`.
2. Allowing addition of types means that some errors will not be caught until later. It's better to catch them at the point they're made.
3. There are many workarounds which are pretty simple.
4. Similarity to splice's type.